samples/net/ocpp/src/main.c

```diff
@@ -0,0 +1,384 @@
+
+			if (io.stop_charge.id_con == idcon)
+				break;
+
+		} while (1);
+	}
+
+	ret = ocpp_stop_transaction(sh, sys_rand32_get(), timeout_ms);
+	if (ret) {
+		LOG_ERR("ocpp stop txn idcon %d> %d\n", idcon, ret);
+		return;
+	}
+
+	LOG_INF("ocpp stop charging connector id %d\n", idcon);
+	k_sleep(K_SECONDS(1));
+	ocpp_session_close(sh);
+	tid[idcon - 1] = NULL;
+	k_sleep(K_SECONDS(1));
+	k_thread_abort(k_current_get());
+}
+
+static int ocpp_getaddrinfo(char *server, int port, char **ip)
+{
+	int ret;
+	uint8_t retry = 5;
+	char addr_str[INET_ADDRSTRLEN];
+	struct sockaddr_storage b;
+	struct addrinfo *result = NULL;
+	struct addrinfo *addr;
+	struct addrinfo hints = {
+		.ai_family = AF_INET,
+		.ai_socktype = SOCK_STREAM
+	};
+
+	LOG_INF("cs server %s %d", server, port);
+	do {
+		ret = getaddrinfo(server, NULL, &hints, &result);
+		if (ret == -EAGAIN) {
+			LOG_ERR("ERROR: getaddrinfo %d, rebind", ret);
+			k_sleep(K_SECONDS(1));
+		} else if (ret) {
+			LOG_ERR("ERROR: getaddrinfo failed %d", ret);
+			return ret;
+		}
+	} while (--retry && ret);
+
+	addr = result;
+	while (addr != NULL) {
+		/* IPv4 Address. */
+		if (addr->ai_addrlen == sizeof(struct sockaddr_in)) {
+			struct sockaddr_in *broker =
+				((struct sockaddr_in *)&b);
+
+			broker->sin_addr.s_addr =
+				((struct sockaddr_in *)addr->ai_addr)
+				->sin_addr.s_addr;
+			broker->sin_family = AF_INET;
+			broker->sin_port = htons(port);
+
+			inet_ntop(AF_INET, &broker->sin_addr, addr_str,
+				  sizeof(addr_str));
+
+			*ip = strdup(addr_str);
+			LOG_INF("IPv4 Address %s", addr_str);
+			break;
+		}
+
+		LOG_INF("error: ai_addrlen = %u should be %u or %u",
+			(unsigned int)addr->ai_addrlen,
+			(unsigned int)sizeof(struct sockaddr_in),
+			(unsigned int)sizeof(struct sockaddr_in6));
+
+		addr = addr->ai_next;
+	}
+
+	/* Free the address. */
+	freeaddrinfo(result);
+
+	return 0;
+}
+
+int main(void)
+{
+	int ret;
+	int i;
+	char *ip = NULL;
+
+	ocpp_cp_info_t cpi = { "basic", "zephyr", .num_of_con = NO_OF_CONN };
+	ocpp_cs_info_t csi = { NULL,
+				"/steve/websocket/CentralSystemService/zephyr",
+				CONFIG_OCPP_PORT,
+				AF_INET };
+
+	printk("OCPP sample %s\n", CONFIG_BOARD);
+
+#if defined(CONFIG_NET_DHCPV4)
+	ocpp_test_dhcp_init();
+#endif
+
+	/* wait for device dhcp ip recive */
+	k_sleep(K_SECONDS(3));
+	ret = ocpp_getaddrinfo(CONFIG_OCPP_SERVER, CONFIG_OCPP_PORT, &ip);
+	if (ret < 0)
+		return ret;
+
+	csi.cs_ip = ip;
+
+	ocpp_get_time_from_sntp();
+
+	ret = ocpp_init(&cpi,
+			&csi,
+			user_notify_cb,
+			NULL);
+	if (ret) {
+		LOG_ERR("ocpp init failed %d\n", ret);
+		return ret;
+	}
+
+	/* Span threads for each connector */
+	for (i = 0; i < NO_OF_CONN; i++) {
+		snprintf(idtag[i], sizeof(idtag[0]), "ZepId%02d", i);
+		tid[i] = k_thread_create(&tinfo[i], cp_stk[i],
+					 sizeof(cp_stk[i]),
+					 ocpp_cp_entry, (void *)(i + 1),
+					 idtag[i], obs[i], 7, 0, K_NO_WAIT);
+	}
+
+	/* Active charging session */
+	k_sleep(K_SECONDS(30));
```

Comment:
I am a bit baffled with these sleeps, so this sample works for 30sec, stops and then sleeps 1200sec. Is there some reason for this behavior?

Reply:
Added comments for reason